ql/src/test/org/apache/hadoop/hive/ql/exec/TestGetPartitionInBatches.java

```diff
@@ -256,4 +258,21 @@ public void testBatchingWhenException() throws Exception {
         // In case any duplicate/incomplete list is given by hive.getAllPartitionsInBatches, the below assertion will fail
         assert(partNames.size() == 0);
     }
+
+    @Test
+    public void testBatchingWhenBatchSizeIsZero() throws MetaException {
+        HiveMetaStoreClient spyMSC = spy(msc);
+        hive.setMSC(spyMSC);
+        int batchSize = 0;
+        try {
+            new PartitionIterable(hive, hive.getTable(dbName, tableName), null, batchSize);
+        } catch (HiveException e) {
+            Assert.assertTrue(e.getMessage().contains("Batch Size for Partition Iterable should be > 0"));
```

Comment:
In this case would need to change the error message.

**It is not belong to your PR:**
btw I feel we overuse the HiveException. I know it is bit out of the scope of this but I have seen several PR the Author checked the message of the exception which is an anti-pattern. Every exception should be clear about the nature of the exception. The message just provides details for the developer. It is not an end-user interface. 

If you are agree (@ayushtkn, @deniskuzZ @abstractdog @zratkai, @veghlaci05 ) , we should have a discussion in the future of the exceptions use in the hive. HiveException is oddly overused.

Reply:
I have just changed the error message here for now.
Though i do agree about the clarity of the nature for every exception.